src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/AndroidCertificatePal.cs

```diff
@@ -0,0 +1,441 @@
+        {
+            get
+            {
+                EnsureCertificateData();
+                return _certData.Issuer;
+            }
+        }
+
+        public PolicyData GetPolicyData()
+        {
+            EnsureCertificateData();
+            PolicyData policyData = default;
+            foreach (X509Extension extension in _certData.Extensions)
+            {
+                switch (extension.Oid!.Value)
+                {
+                    case Oids.ApplicationCertPolicies:
+                        policyData.ApplicationCertPolicies = extension.RawData;
+                        break;
+                    case Oids.CertPolicies:
+                        policyData.CertPolicies = extension.RawData;
+                        break;
+                    case Oids.CertPolicyMappings:
+                        policyData.CertPolicyMappings = extension.RawData;
+                        break;
+                    case Oids.CertPolicyConstraints:
+                        policyData.CertPolicyConstraints = extension.RawData;
+                        break;
+                    case Oids.EnhancedKeyUsage:
+                        policyData.EnhancedKeyUsage = extension.RawData;
+                        break;
+                    case Oids.InhibitAnyPolicyExtension:
+                        policyData.InhibitAnyPolicyExtension = extension.RawData;
+                        break;
+                }
+            }
+
+            return policyData;
+        }
+
+        public IEnumerable<X509Extension> Extensions
+        {
+            get
+            {
+                EnsureCertificateData();
+                return _certData.Extensions;
+            }
+        }
+
+        internal void SetPrivateKey(SafeEvpPKeyHandle privateKey)
+        {
+            _privateKey = privateKey;
+        }
+
+        internal SafeEvpPKeyHandle? PrivateKeyHandle
+        {
+            get { return _privateKey; }
+        }
+
+        public RSA? GetRSAPrivateKey()
+        {
+            if (_privateKey == null || _privateKey.IsInvalid)
+            {
+                return null;
+            }
+
+            return new RSAOpenSsl(_privateKey);
+        }
+
+        public DSA? GetDSAPrivateKey()
+        {
+            if (_privateKey == null || _privateKey.IsInvalid)
+            {
+                return null;
+            }
+
+            return new DSAOpenSsl(_privateKey);
+        }
+
+        public ECDsa GetECDsaPublicKey()
+        {
+            throw new NotImplementedException(nameof(GetECDsaPublicKey));
+        }
+
+        public ECDiffieHellman GetECDiffieHellmanPublicKey()
+        {
+            throw new NotImplementedException(nameof(GetECDiffieHellmanPublicKey));
+        }
+
+        public ECDsa? GetECDsaPrivateKey()
+        {
+            if (_privateKey == null || _privateKey.IsInvalid)
+            {
+                return null;
+            }
+
+            return new ECDsaOpenSsl(_privateKey);
```

Comment:
We'll want to make sure to coordinate with my ECC PR to update the usages here with the new Android types for these algorithms.

Reply:
Good call. I'll switch these to just throw `NotImplemented` for this change.